rspace/src/main/scala/coop/rchain/rspace/RSpace.scala

```diff
@@ -210,7 +210,8 @@ class RSpace[F[_]: Async: Log: Metrics: Span, C, P, A, K](
 
   def spawn: F[ISpace[F, C, P, A, K]] = spanF.withMarks("spawn") {
     for {
-      historyRepo   <- Sync[F].delay(historyRepositoryAtom.get())
+      historyRef    <- historyRepositoryRef
+      historyRepo   <- historyRef.get
```

Comment:
I can see here that previously shared `historyRepositoryAtom` has been used, but now each such call (in other places as well) creates new instance of Ref. 

Could you try in RSpaceOps 
```scala
protected val historyRepositoryRef: F[Ref[F, HistoryRepository[F, C, P, A, K]]] =
    Ref.unsafe[F, HistoryRepository[F, C, P, A, K]](historyRepository)
```
instead of
```scala
protected val historyRepositoryRef: F[Ref[F, HistoryRepository[F, C, P, A, K]]] =
    Ref.of[F, HistoryRepository[F, C, P, A, K]](historyRepository)
```
and access to this shared object everywhere?

I'm interested to see whether this will break some tests. Since this is what I tried myself before.